src/components/testCaseLanding/qdm/TestCaseCoverage/CoverageTabs/CoverageTabList.tsx

```diff
@@ -1,110 +1,115 @@
-  calculationOutput,
+  cqlDefinitionCallstack,
+  testCaseGroups,
 }: Props) => {
-  const getDefintionCategoryFilteringCondition = (
-    definitionType,
-    statementResult,
-    definitionName
-  ) => {
-    if (definitionType === "Used") {
-      return statementResult[definitionName].relevance !== "NA";
-    }
-    if (definitionType === "Unused") {
-      return statementResult[definitionName].relevance === "NA";
+  const [relevantPops, setRelevantPops] = useState([]);
+  useEffect(() => {
+    if (testCaseGroups && populationCriteria.id) {
+      const selectedGroup = testCaseGroups.find(
+        (gp) => gp.groupId === populationCriteria.id
+      );
+      const results = selectedGroup?.populationValues
+        .filter((population) => {
+          return !population.name.includes("Observation");
+        })
+        .map((population, index) => {
+          return {
+            id: population.id,
+            criteriaReference: population.criteriaReference,
+            name: population.name,
+            abbreviation: getPopulationAbbreviation(
+              selectedGroup.populationValues,
+              population.name,
+              index
+            ),
+          };
+        });
+      setRelevantPops(results);
     }
-  };
-
-  const filterBasedOnDefinitionCategories = (definitionType) => {
-    const definitionNames = [
-      ...new Set(
-        Object.values(calculationOutput)?.flatMap((testCases) =>
-          Object.values(testCases)?.flatMap((testCaseCalculationResult) =>
-            Object.values(testCaseCalculationResult.statement_results)?.flatMap(
-              (statementResult) =>
-                Object.keys(statementResult)?.filter((definitionName) =>
-                  getDefintionCategoryFilteringCondition(
-                    definitionType,
-                    statementResult,
-                    definitionName
-                  )
-                )
-            )
-          )
-        )
-      ),
-    ];
+  }, [testCaseGroups, populationCriteria.id, getPopulationAbbreviation]);
 
-    return definitionNames.reduce((acc, definitionName) => {
-      if (mappedCql.definitions[definitionName]) {
-        acc[definitionName] = mappedCql.definitions[definitionName];
+  const getStatementResultsInCategory = (statementResults, definition) => {
+    if (statementResults) {
+      if (definition === "Used") {
+        return statementResults.filter((s) => s.relevance !== "NA");
       }
-      return acc;
-    }, {});
-  };
-
-  const getDefinitionResults = (definitionType) => {
-    if (
-      mappedCql &&
-      mappedCql.functions &&
-      mappedCql.definitions &&
-      calculationOutput
-    ) {
-      if (definitionType === "Functions") {
-        return mappedCql.functions;
-      }
-      if (definitionType === "Unused") {
-        return filterBasedOnDefinitionCategories(definitionType);
+      if (definition === "Unused") {
+        return statementResults.filter((s) => s.relevance === "NA");
```

Comment:
need to ignore functions here. let's add `s.type != 'FunctionDef'` condition here